core/src/main/java/ai/timefold/solver/core/config/heuristic/selector/move/NearbyConstructionHeuristicAutoConfigurationMoveSelectorConfig.java

```diff
@@ -0,0 +1,19 @@
+package ai.timefold.solver.core.config.heuristic.selector.move;
+
+import java.util.Random;
+
+import ai.timefold.solver.core.impl.heuristic.selector.common.nearby.NearbyDistanceMeter;
+
+/**
+ * General superclass for move selectors that support Nearby Selection autoconfiguration in construction heuristics.
+ */
+public abstract class NearbyConstructionHeuristicAutoConfigurationMoveSelectorConfig<Config_ extends MoveSelectorConfig<Config_>>
```

Comment:
I'm unsure if it is a good idea for this interface to extend `NearbyAutoConfigurationMoveSelectorConfig`. I am aware that multiple inheritance is not supported, so I suggest converting the classes `NearbyConstructionHeuristicAutoConfigurationMoveSelectorConfig` and `NearbyAutoConfigurationMoveSelectorConfig` into interfaces and making `ListChangeMoveSelectorConfig` and `ChangeMoveSelectorConfig` implement both contracts.

I believe this will improve the readability, and `NearbyConstructionHeuristicAutoConfigurationMoveSelectorConfig` won't directly result in automatically enabling nearby for the local search moves.

Maybe it makes sense to rename `NearbyAutoConfigurationMoveSelectorConfig` to `NearbyLocalSearchAutoConfigurationMoveSelectorConfig`.